File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/FloatConverter.java
package org.matilda.commands.types;

import com.google.protobuf.Any;
import com.google.protobuf.FloatValue;
import com.google.protobuf.InvalidProtocolBufferException;

public class FloatConverter implements ProtobufConverter<Float> {
    @Override
    public FloatValue convertToProtobuf(Float object) {
        return FloatValue.newBuilder().setValue(object).build();
    }

    @Override
    public Float convertFromProtobuf(Any object) throws InvalidProtocolBufferException {
        return object.unpack(FloatValue.class).getValue();
    }
}

File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/FloatConverter.kt
package org.matilda.commands.types

import com.google.protobuf.Any
import com.google.protobuf.FloatValue

class FloatConverter : ProtobufConverter<Float> {
    override fun convertToProtobuf(obj: Float): FloatValue {
        return FloatValue.newBuilder().setValue(obj).build()
    }

    override fun convertFromProtobuf(obj: Any): Float {
        return obj.unpack(FloatValue::class.java).value
    }
}
